orchestra/static/orchestra/task/task.controller.es6.js

```diff
@@ -61,9 +61,11 @@ export default function TaskController (
     })
   }
 
-  vm.confirmSubmission = function (command) {
+  vm.confirmSubmission = async function (command) {
     vm.submitting = true
-    if (orchestraService.signals.fireSignal('submit.before') === false) {
+    const submitBefore = await orchestraService.signals.fireSignal(
+      'submit.before')
+    if (submitBefore === false) {
```

Comment:
above you inlined the await inside the if (line 57 of autosave) --- could do the same thing here for consistency